Sources_v3/Database/DatabaseContainer.swift

```diff
@@ -201,14 +223,30 @@ class DatabaseContainer: NSPersistentContainer {
         }
     }
     
-    /// Recreate rersistent store
+    /// Removes the loaded persistent store and tries to recreate it.
     func recreatePersistentStore() throws {
-        guard let url = persistentStoreDescriptions.first!.url
-        else { throw ClientError("Internal Error. Wrong database file url.") }
+        log.assert(
+            persistentStoreDescriptions.count == 1,
+            "DatabaseContainer always assumes 1 persistent store description. Existing descriptions: \(persistentStoreDescriptions)"
+        )
         
-        try persistentStoreCoordinator.destroyPersistentStore(at: url, ofType: "sqlite", options: nil)
-        try FileManager.default.removeItem(at: url)
+        guard let storeDescription = persistentStoreDescriptions.first else {
+            throw ClientError("No persisten store descriptions available.")
+        }
+
+        // Remove all loaded persistent stores first
+        try persistentStoreCoordinator.persistentStores.forEach { store in
+            try persistentStoreCoordinator.remove(store)
+        }
         
+        // If the store was SQLite store, remove the actual DB file
+        if storeDescription.type == NSSQLiteStoreType,
+            let storeURL = storeDescription.url,
+            storeURL.absoluteString.hasSuffix("/dev/null") == false {
```

Comment:
Seems like now we can remove `kind != .inMemory` check from `init` as we are safe inside `recreatePersistentStore()`.

Reply:
We can but it's not necessary. The check is technically correct. There's nothing to recreate if we use in-memory store :)

Reply:
Fixed in 92cbe57af5ba29cb17434a9926d0d481d50d3b51